exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/ConvertCountToDirectScan.java

```diff
@@ -103,6 +104,10 @@ public void onMatch(RelOptRuleCall call) {
       return;
     }
 
+    if (proj != null && !ProjectRemoveRule.isTrivial(proj)) {
```

Comment:
I have a patch, which works fine for Jacque's new unit test. It continues to use directScan for simple count query. The patch is pending pre-commit & unit test run. Will update results shortly.